warehouse/ingest-core/src/test/java/datawave/ingest/table/aggregator/GlobalIndexUidAggregatorExpandedTest.java

```diff
@@ -0,0 +1,719 @@
+        // information?
+        // Note this tests both orderings of values (forward and reverse).
+        // 1. either start with a count-only and add two uids
+        // 2. or start with two named uids and add a count-only
+        agg = new GlobalIndexUidAggregator(2);
+        Value value1 = UidTestBuilder.newBuilder().withCountOnly(3).build();
+        
+        Value value2 = UidTestBuilder.newBuilder().withUids("uid3", "uid4").build();
+        
+        Uid.List expectation = UidTestBuilder.valueToUidList(UidTestBuilder.newBuilder().withCountOnly(5).build());
+        
+        testCombinations(expectation, asList(value1, value2));
+    }
+    
+    @Test
+    public void exceedMaxAddTwoDropOne() {
+        // Will flipping to seen ignore take into account the available information?
+        // Note this tests both orderings of values.
+        // i.e., count = this.uids.size() - this.uidsToRemove.size();
+        agg = new GlobalIndexUidAggregator(2);
+        Value value1 = UidTestBuilder.newBuilder().withCountOnly(3).build();
+        
+        Value value2 = UidTestBuilder.newBuilder().withUids("uid3", "uid4").build();
+        
+        Value value3 = UidTestBuilder.newBuilder().withUids().withRemovals("uid5").build();
+        
+        Uid.List expectation = UidTestBuilder.valueToUidList(UidTestBuilder.newBuilder().withCountOnly(4).build());
+        
+        testCombinations(expectation, asList(value1, value2, value3));
+    }
+    
+    @Test
+    public void removalToZeroCount() {
+        // Will flipping to seen ignore take into account the available information?
+        // Note this tests both orderings of values.
+        // i.e., count = this.uids.size() - this.uidsToRemove.size();
+        agg = new GlobalIndexUidAggregator(10);
+        Value value1 = UidTestBuilder.newBuilder().withCountOnly(1).build();
+        
+        Value value2 = UidTestBuilder.newBuilder().withUids().withRemovals("uid1").build();
+        
+        Uid.List expectation = UidTestBuilder.valueToUidList(UidTestBuilder.newBuilder().withCountOnly(0).build());
+        
+        testCombinations(expectation, asList(value1, value2));
+    }
+    
+    @Test
+    public void countOnlyReduced() {
+        // Will flipping to seen ignore take into account the available information?
+        // Note this tests both orderings of values.
+        // i.e., count = this.uids.size() - this.uidsToRemove.size();
+        agg = new GlobalIndexUidAggregator(10);
+        Value value1 = UidTestBuilder.newBuilder().withCountOnly(30).build();
+        
+        Value value2 = UidTestBuilder.newBuilder().withUids().withRemovals("uid1", "uid2").build();
+        
+        Uid.List expectation = UidTestBuilder.valueToUidList(UidTestBuilder.newBuilder().withCountOnly(28).build());
+        
+        testCombinations(expectation, asList(value1, value2));
+    }
+    
+    private void testCombinations(Uid.List expectation, List<Value> input) {
```

Comment:
Might not be worth changing everywhere, but you could have input be a varargs parameter and do the asList call internally to this method. It would make all of the calls to this just a touch more readable.

Reply:
I made this change as recommended.